src/rax.c

```diff
@@ -499,6 +499,92 @@ static inline size_t raxLowWalk(rax *rax, unsigned char *s, size_t len, raxNode
     return i;
 }
 
+int raxIteratorPushChildOffset(raxIterator *it, uint8_t child_offset) {
+    if (it->cpos == it->cpos_max) {
+        uint8_t *old = (
+            (it->child_offset_stack == it->child_offset_stack_static) ?
+            NULL :
+            it->child_offset_stack
+        );
+
+        size_t new_max = it->cpos_max * 2;
+        it->child_offset_stack = rax_realloc(old, new_max);
+
+        if (it->child_offset_stack == NULL) {
+            it->child_offset_stack = (!old) ? it->child_offset_stack_static : old;
+            errno = ENOMEM;
+            return 0;
+        }
+
+        if (old == NULL)
+            memcpy(it->child_offset_stack, it->child_offset_stack_static, it->cpos_max);
+
+        it->cpos_max = new_max;
+    }
+
+    it->child_offset_stack[it->cpos] = child_offset;
+    it->cpos++;
+    return 1;
+}
+
+ uint8_t raxIteratorPopChildOffset(raxIterator *it) {
+    if (it->cpos == 0) return 0;
+    it->cpos--;
+    return it->child_offset_stack[it->cpos];
+}
+
+/* See cmts in raxLowWalk() above */
+static inline size_t raxLowWalkSeek(
+    unsigned char *s, size_t len, int *splitpos, raxIterator* it
+) {
```

Comment:
`raxIterator* it` should be the first param.
```suggestion
static inline size_t raxLowWalkSeek(raxIterator* it, unsigned char *s,
                                    size_t len, int *splitpos)
{
```

Reply:
I wonder why this method was added? Why not just modify the `raxLowWalk`.

Reply:
I tried that but it slowed performance, as I recall, of e.g. raxFind(). But hopefully my approach can be improved.

Reply:
Have you confirmed why the modifications to raxLowWalk would cause performance degradation?